Zscaler/zscaler-zia/ingest/parser.yml

```
@@ -0,0 +1,122 @@
+name: withsecure-elements
+pipeline:
+  - name: json_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: "{{original.message}}"
+        output_field: message
+
+  - name: parse_datetime
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{json_event.message.event.datetime}}"
+        output_field: result
+
+  - name: parse_time
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{json_event.message.event.time}}"
+        output_field: result
+
+  - name: set_common_fields
+  - name: set_event_category
+
+stages:
+  set_event_category:
+    actions:
+      - set:
+          event.category: ["process"]
+          event.type: ["info"]
+
+      - set:
+          event.category: ["network"]
+          event.type: ["info"]
+        filter: "{{json_event.message.get('sourcetype') == 'zscalernss-web' or json_event.message.get('sourcetype') == 'zscalernss-dns'}}"
+
+      - set:
+          event.category: ["network"]
+          event.type: ["connection"]
+        filter: "{{json_event.message.get('sourcetype') == 'zscalernss-fw'}}"
+
+      - set:
+          event.category: ["authentication"]
+          event.type: ["start"]
+        filter: "{{json_event.message.get('sourcetype') == 'zscalernss-audit' and json_event.message.event.action.lower() == 'sign_in'}}"
+
+      - set:
+          event.category: ["configuration"]
+          event.type: ["change"]
+        filter: "{{json_event.message.get('sourcetype') == 'zscalernss-audit' and json_event.message.event.action.lower() == 'update'}}"
+
+  set_common_fields:
+    actions:
+      - set:
+          "@timestamp": "{{parse_datetime.result or parse_time.result}}"
+        filter: "{{parse_datetime.result != None}}"
+
+      - set:
+          "@timestamp": "{{parse_time.result}}"
+        filter: "{{parse_time.result != None}}"
+
+      - set:
+          event.kind: "event"
+          event.duration: "{{json_event.message.event.duration}}"
+          event.action: "{{json_event.message.event.action | lower}}"
+
+          zscaler.zia.event_id: "{{json_event.message.event.event_id}}"
+          zscaler.zia.source_type: "{{json_event.message.sourcetype}}"
+          zscaler.zia.action.result: "{{json_event.message.event.result}}"
+          zscaler.zia.vendor: "{{json_event.message.event.vendor}}"
+          zscaler.zia.product: "{{json_event.message.event.product}}"
+          zscaler.zia.appname: "{{json_event.message.event.appname}}"
+          zscaler.zia.category: "{{json_event.message.event.category}}"
+          zscaler.zia.sub_category: "{{json_event.message.event.subcategory}}"
+          zscaler.zia.department: "{{json_event.message.event.department}}"
+          zscaler.zia.audit.log_type: "{{json_event.message.event.auditlogtype}}"
```

I suggest extracting the resource (for audit message).

```suggestion
          zscaler.zia.audit.log_type: "{{json_event.message.event.auditlogtype}}"
          zscaler.zia.resource: "{{json_event.message.event.resource}}"
```